src/v/rpc/rpc_compiler.py

```diff
@@ -56,12 +56,12 @@
 namespace {{namespace}} {
 
 template<typename Codec>
-class {{service_name}}_service_base : public rpc::service {
+class {{service_name}}_service_base : public ::rpc::service {
```

Comment:
well I have an RPC service that is wrapped in the namespace `transform::rpc`. So now anything qualified with `rpc` will lookup `transform::rpc::*`.

We have to either:
1) choose a different name for the namespace instead of `transform::rpc`. I am open to different names.
2) do this and fully qualify the name here.

I don't see another option unless I'm missing something? We've never needed this before because nobody else has tried to use an `rpc` in a nested namespace. I'm probably being overly picky, but I liked the classes introduced in this PR being namespaced as `transform::rpc::client` and `transform::rpc::service`. But I can drop this commit and use a different namespace like `data`, `io` or I guess not nest the namespace. My instinct is that each directory is a CMake target and it's own namespace as well, but I could keep it in the transform namespace (or choose another one).

Reply:
>We've never needed this before because nobody else has tried to use an rpc in a nested namespace

makes sense. i think i missed this originally